timemachine/fe/rbfe.py

```diff
@@ -98,7 +99,24 @@ def sample(initial_state, protocol):
     intg_impl = initial_state.integrator.impl()
     baro_impl = initial_state.barostat.impl(bound_impls)
 
-    ctxt = custom_ops.Context(initial_state.x0, initial_state.v0, initial_state.box0, intg_impl, bound_impls, baro_impl)
+    # minimize the local region, removing clashes and unstretching some bonds
+    cutoff = 1.0  # in nanometers
+    # local minimize region around ligand
+    ligand_coords = initial_state.x0[initial_state.ligand_idxs]
+    d_ij = cdist(ligand_coords, initial_state.x0)
+    # if any atom is within any of the ligand's atom's ixn radius, flag it for minimization
+    free_idxs = np.where(np.any(d_ij < cutoff, axis=0))[0].tolist()
+
+    val_and_grad_fn = minimizer.get_val_and_grad_fn(initial_state.potentials, initial_state.box0, initial_state.lamb)
+
+    assert np.all(np.isfinite(initial_state.x0)), "Initial coordinates contain nan or inf"
+
+    x0_min = minimizer.local_minimize(initial_state.x0, val_and_grad_fn, free_idxs)
+
+    assert not np.any(np.isnan(x0_min)), "Minimization resulted in a nan"
```

Comment:
nit: any reason not to assert all finite here (as opposed to just no NaNs) as above?

Reply:
willfix (forgot to copy the above test)